protocol/x/clob/types/message_batch_cancel_test.go

```diff
@@ -0,0 +1,123 @@
+package types_test
+
+import (
+	"testing"
+
+	"github.com/dydxprotocol/v4-chain/protocol/testutil/constants"
+	"github.com/dydxprotocol/v4-chain/protocol/x/clob/types"
+	satypes "github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMsgBatchCancel_ValidateBasic(t *testing.T) {
+	oneOverMax := []uint32{}
+	for i := uint32(0); i < types.MaxMsgBatchCancelBatchSize+1; i++ {
+		oneOverMax = append(oneOverMax, i)
+	}
+
+	tests := map[string]struct {
+		msg types.MsgBatchCancel
+		err error
+	}{
+		"invalid subaccount": {
+			msg: *types.NewMsgBatchCancel(
+				constants.InvalidSubaccountIdNumber,
+				[]types.OrderBatch{
+					{
+						ClobPairId: 0,
+						ClientIds: []uint32{
+							0, 1, 2, 3,
+						},
+					},
+				},
+				10,
+			),
+			err: satypes.ErrInvalidSubaccountIdNumber,
+		},
+		"over 100 cancels in for one clob pair id": {
+			msg: *types.NewMsgBatchCancel(
+				constants.Alice_Num0,
+				[]types.OrderBatch{
+					{
+						ClobPairId: 0,
+						ClientIds:  oneOverMax,
+					},
+				},
+				10,
+			),
+			err: types.ErrInvalidBatchCancel,
+		},
+		"over 100 cancels split over two clob pair id": {
+			msg: *types.NewMsgBatchCancel(
+				constants.Alice_Num0,
+				[]types.OrderBatch{
+					{
+						ClobPairId: 0,
+						ClientIds:  oneOverMax[:52],
```

Comment:
meganit: instead of hardcoding 52 here, this should be a function of MaxShortTermCancels which is subject to change